codegen/src/main/resources/META-INF/proguard/dagger-reflect.pro

```diff
@@ -0,0 +1,49 @@
+# When using factories or builders, the enclosing class is used to determine the component type.
```

Comment:
I don't see a mention of `@Binds` which I think may be a special case as it's not referenced by generated Dagger component code. Is it covered by one of these rules?

Reply:
These rules are for the codegen artifact reflection, not the reflection in the reflect artifact.

I'm not convinced anyone will be using shrinking with the reflect artifact. I'd be curious what their use case for this was.

Reply:
Oh, good point, sorry.

Re shrinking with reflect: we sometimes minifyEnabled=true on debug flavor so we can test shrinking-related issues. If dagger-reflect was set up it with `if (idea)` this quick-change-for-debugging would fail the build unexpectedly because of reflect.

Reply:
In that case, though, having dagger-reflect in your APK would already produce results that were not representative of what happens when shrinking in release. In release builds a large portion of Dagger-generated types are vertically merge and/or inlined. I've even written some non-trivial graphs that have been _completely_ eliminated by shrinking leaving only a bunch of chained calls to constructors.

I'm not opposed to adding rules to the reflect artifact. They're probably pretty basic. But I think the concept of shrinking and using reflect are mutually exclusive in their goals so it's not a high priority.

Reply:
What I was referring to here is adding `minifyEnabled=true` to debug so that we can verify things not related to Dagger, or dagger-reflect. But at the moment if the debug build is set up with dagger-reflect, it prevents debugging other problems., because injection crashes. I see why it's not high priority though, as it shouldn't be hard to turn off dagger-reflect, it's just inconvenient.